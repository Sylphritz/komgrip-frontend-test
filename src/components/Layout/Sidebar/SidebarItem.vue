<template>
  <li class="nav-item">
    <a class="nav-link" :class="{ active: isActive }" :href="url">
      <div
        class="icon icon-shape icon-sm shadow border-radius-md bg-white text-center me-2 d-flex align-items-center justify-content-center"
      >
        <Component v-if="icon" :is="icon" />
        <i v-else class="fas fa-square"></i>
      </div>
      <span class="nav-link-text ms-1">
        <slot />
      </span>
    </a>
  </li>
</template>

<script setup lang="ts">
defineProps({
  url: {
    type: String,
    required: true
  },
  icon: {
    type: Object
  },
  isActive: {
    type: Boolean
  }
})
</script>

<style lang="scss" scoped>
.fas {
  color: inherit;
}
</style>
