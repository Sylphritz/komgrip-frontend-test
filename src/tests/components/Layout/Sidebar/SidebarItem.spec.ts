import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils'
import SidebarItem from '@/components/Layout/Sidebar/SidebarItem.vue'
import ShopIcon from '@/components/Icons/ShopIcon.vue'

describe('SidebarItem', () => {
  const url = 'testUrl'

  it('renders successfully and matches the snapshot', () => {
    const wrapper = mount(SidebarItem, {
      props: {
        url,
        icon: ShopIcon,
        isActive: false
      },
      shallow: true
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  it('links to the correct url', () => {
    const wrapper = mount(SidebarItem, {
      props: {
        url,
        icon: ShopIcon,
        isActive: false
      },
      shallow: true
    })
    expect(wrapper.find('a.nav-link').attributes('href')).toBe(url)
  })

  it('has active class when set to active', () => {
    const wrapper = mount(SidebarItem, {
      props: {
        url,
        icon: ShopIcon,
        isActive: true
      },
      shallow: true
    })

    expect(wrapper.find('a.nav-link').classes()).toStrictEqual(['nav-link', 'active'])
  })

  it('renders the default icon when no icon is supplied', () => {
    const wrapper = mount(SidebarItem, {
      props: {
        url,
        isActive: false
      },
      shallow: true
    })

    expect(wrapper.find('.fas.fa-square')).toBeDefined()
  })
})
