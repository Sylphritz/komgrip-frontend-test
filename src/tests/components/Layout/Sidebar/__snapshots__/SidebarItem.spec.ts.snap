// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SidebarItem > renders successfully and matches the snapshot 1`] = `
<li
  class="nav-item"
  data-v-b6f5b118=""
>
  <a
    class="nav-link"
    data-v-b6f5b118=""
    href="testUrl"
  >
    <div
      class="icon icon-shape icon-sm shadow border-radius-md bg-white text-center me-2 d-flex align-items-center justify-content-center"
      data-v-b6f5b118=""
    >
      <anonymous-stub
        data-v-b6f5b118=""
      />
    </div>
    <span
      class="nav-link-text ms-1"
      data-v-b6f5b118=""
    >
      
      
    </span>
  </a>
</li>
`;
